updateRepo.sh

```diff
@@ -1,34 +1,68 @@
+#!/bin/bash
+set -ex
+
 echo "Checking if repo needs to be updated"
-export ORG="IBM-Swift"
-export REPO="generator-swiftserver-projects"
-export GH_REPO="github.com/${ORG}/${REPO}.git"
-export BRANCH="init"
-export projectName="Generator-Swiftserver-Projects"
-
-cd ${TRAVIS_BUILD_DIR}
-mkdir current
-cd current
-git clone -b ${BRANCH} "https://${GITHUB_USERNAME}:${GITHUB_PASSWORD}@github.com/${ORG}/${REPO}.git"
-export currentProject=`pwd`
-
-mkdir -p ../new/${projectName}
-cd ../new/${projectName}
-yo swiftserver --init --skip-build
-echo "Generate README rtf"
-pandoc README.md -f markdown_github -t rtf -so README.rtf
-cd ../
-export newProject=`pwd`
-
-if diff -x '.git' -r ${currentProject}/${REPO} ${newProject}/${projectName}
-then
-  echo "Project does not need to be updated"
-  rm -rf ${currentProject}/${REPO} ${newProject}/${projectName}
-  exit 1
-fi
-
-echo "Project needs to be updated"
-cp -r ${newProject}/${projectName}/. ${currentProject}/${REPO}
-cd ${currentProject}/${REPO}
-git add .
-git commit -m "CRON JOB: Updating generated project"
-git push origin ${BRANCH}
+ORG="IBM-Swift"
+REPO="generator-swiftserver-projects"
+GH_REPO="github.com/${ORG}/${REPO}.git"
+BRANCHES="init openAPI"
+projectName="Generator-Swiftserver-Projects"
+
+SUCCESS="" # List of successful updates
+FAIL=""    # List of failed updates
+
+# Builds a list of branchs that failed to update (if any)
+function fail () {
+  FAIL="$FAIL $1" && echo "Failed to push to branch: $BRANCH"
+  return 1
+}
+
+for BRANCH in $BRANCHES
+do
+  echo "Generating project for ${BRANCH}"
+  cd "${TRAVIS_BUILD_DIR}"
+  rm -rf current
+  rm -rf new
+  git clone -b "${BRANCH}" "https://${GITHUB_USERNAME}:${GITHUB_PASSWORD}@github.com/${ORG}/${REPO}.git" current
+  currentProject="$(pwd)/current"
+
+  # Need to create a project directory and move into it so we can run the generator.
+  mkdir -p "new/${projectName}" && cd "new/${projectName}"
+  if [[ "${BRANCH}" == "init" ]]
+  then
+    yo swiftserver --init --skip-build
+  elif [[ "$BRANCH" == "openAPI" ]]
+  then
+    yo swiftserver --app --spec '{ "appName":"'"$projectName"'", "appType":"scaffold", "appDir":".", "openapi":true, "docker":true, "metrics":true, "healthcheck":true }' --skip-build
+    rm spec.json
+  else
+    exit
+  fi
+  # Step back into the travis build directory after generator has finished.
+  cd "${TRAVIS_BUILD_DIR}"
+
+  echo "Generate README rtf"
+  pandoc README.md -f markdown_github -t rtf -so README.rtf
+  newProject="$(pwd)/new"
+
+  currentRepo="${currentProject}/${REPO}"
+  newRepo="${newProject}/${projectName}"
+
+  if diff -x '.git' -r "${currentRepo}" "${newRepo}"
+  then
+    echo "Project does not need to be updated"
+    rm -rf "${currentRepo}" "${newRepo}"
+    exit 1
```

Comment:
This doesn't seem right - shouldn't this be `continue`?
You may have other repos in the enclosing `for` loop to try.  Another symptom is that the final output from the script "Success: <list of branches" doesn't appear in your build log: https://travis-ci.org/IBM-Swift/generator-swiftserver-projects/builds/511976963